MCR-1396 IIIF-Presentation-API

sebhofmann opened the pull request:
Added first draft of IIIF-Presentation-API

sebhofmann committed:
MCR-1396 IIIF-Presentation-API

Added first draft of IIIF-Presentation-API

rsteph-de reviewed (approved):
please set MyCoRe/GPL header comment before merging this component

sebhofmann committed:
MCR-1396 IIIF-Presentation-API

added MyCoRe GPL header

sebhofmann committed:
MCR-1396 IIIF-Presentation-API

removed getCollection method

yagee-de commented on mycore-iiif/src/main/java/org/mycore/iiif/image/resources/MCRIIIFImageResource.java:
```diff
@@ -0,0 +1,172 @@
+
+    @GET
+    @Produces("application/ld+json")
+    @Path("{impl}/{identifier}/info.json")
```
every method starts with `impl` path parameter. Should be path parameter of class

yagee-de commented on mycore-iiif/src/main/java/org/mycore/iiif/presentation/resources/MCRIIIFPresentationResource.java:
```diff
@@ -0,0 +1,173 @@
+
+    @GET
+    @Produces(APPLICATION_LD_JSON)
+    @Path("{impl}/{identifier}/manifest")
```
every method starts with `impl` path parameter. Should be path parameter of class

yagee-de committed:
MCR-1396 IIIF Presentation API

Added mycore-iiif to mycore-meta

refactored code:
 - moved {impl} PathParam to class
 - set common param names as fields
 - add MCRIIIFMediaType class

yagee-de committed:
MCR-1396 IIIF Presentation API

Added mycore-iiif to mycore-meta

refactored code:
 - moved {impl} PathParam to class
 - set common param names as fields

yagee-de commented:
:shipit: ?

sebhofmann commented:
:thumbsup:
